pkg/collector/collector.go

```diff
@@ -1,228 +1,19 @@
-				clientExpiry,
-				prometheus.GaugeValue,
-				float64(ci.ChainAClientExpiration.Unix()),
-				[]string{
-					(*cc.RPCs)[path.Chain1.ChainName].ChainID,
-					path.Chain1.ClientID,
-					(*cc.RPCs)[path.Chain2.ChainName].ChainID,
-					discordIDs,
-					status,
-				}...,
-			)
-
-			ch <- prometheus.MustNewConstMetric(
-				clientExpiry,
-				prometheus.GaugeValue,
-				float64(ci.ChainBClientExpiration.Unix()),
-				[]string{
-					(*cc.RPCs)[path.Chain2.ChainName].ChainID,
-					path.Chain2.ClientID,
-					(*cc.RPCs)[path.Chain1.ChainName].ChainID,
-					discordIDs,
-					status,
-				}...,
-			)
-
-			// Stuck packets
-			status = successStatus
-
-			stuckPackets, err := ibc.GetChannelsInfo(path, cc.RPCs)
-			if err != nil {
-				status = errorStatus
-
-				log.Error(err.Error())
-			}
-
-			if !reflect.DeepEqual(stuckPackets, ibc.ChannelsInfo{}) {
-				for _, sp := range stuckPackets.Channels {
-					ch <- prometheus.MustNewConstMetric(
-						channelStuckPackets,
-						prometheus.GaugeValue,
-						float64(sp.StuckPackets.Source),
-						[]string{
-							sp.Source,
-							sp.Destination,
-							(*cc.RPCs)[path.Chain1.ChainName].ChainID,
-							(*cc.RPCs)[path.Chain2.ChainName].ChainID,
-							path.Chain1.ChainName,
-							path.Chain2.ChainName,
-							discordIDs,
-							status,
-						}...,
-					)
-
-					ch <- prometheus.MustNewConstMetric(
-						channelStuckPackets,
-						prometheus.GaugeValue,
-						float64(sp.StuckPackets.Destination),
-						[]string{
-							sp.Destination,
-							sp.Source,
-							(*cc.RPCs)[path.Chain2.ChainName].ChainID,
-							(*cc.RPCs)[path.Chain1.ChainName].ChainID,
-							path.Chain2.ChainName,
-							path.Chain1.ChainName,
-							discordIDs,
-							status,
-						}...,
-					)
-				}
-			}
-		}(p)
-	}
-
-	wg.Wait()
-
-	log.Debug("Stop collecting", zap.String("metric", clientExpiryMetricName))
-}
-
-func (wb WalletBalanceCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- walletBalance
-}
-
-func (wb WalletBalanceCollector) Collect(ch chan<- prometheus.Metric) {
-	log.Debug("Start collecting", zap.String("metric", walletBalanceMetricName))
-
-	var wg sync.WaitGroup
-
-	for _, a := range wb.Accounts {
-		wg.Add(1)
-
-		go func(account config.Account) {
-			defer wg.Done()
-
-			balance := 0.0
-			status := successStatus
-
-			err := account.GetBalance(wb.RPCs)
-			if err != nil {
-				status = errorStatus
-
-				log.Error(err.Error(), zap.Any("account", account))
-			} else {
-				// Convert to a big float to get a float64 for metrics
-				balance, _ = big.NewFloat(0.0).SetInt(account.Balance.BigInt()).Float64()
-			}
-
-			ch <- prometheus.MustNewConstMetric(
-				walletBalance,
-				prometheus.GaugeValue,
-				balance,
-				[]string{account.Address, (*wb.RPCs)[account.ChainName].ChainID, account.Denom, status}...,
-			)
-		}(a)
-	}
-
-	wg.Wait()
-
-	log.Debug("Stop collecting", zap.String("metric", walletBalanceMetricName))
-}
+var ctx = context.Background()
```

Comment:
Why to create a package global variable? Isn't it better to add `ctx` property to each collector struct?

Reply:
did this to please the linter :) I did consider collector level context but at the end the separation is only code files but the other variables are also global to package level e.g. clientExpiry. Linter liked this way better, is it a blocker?

Reply:
up to you, doesn't matter much